app/controllers/tickets_controller.rb

```diff
@@ -1,6 +1,43 @@
 # frozen_string_literal: true
 
 class TicketsController < ApplicationController
+  include Rails::Pagination
+
+  SORT_WEIGHT_LAMBDAS = {
+    createdAt:
+      lambda { |ticket| ticket.created_at },
+  }.freeze
+
+  def index
+    tickets = Ticket
+
+    # Filter based on params
+    type = params[:type]
+    if type
+      tickets = tickets.where(metadata_type: type)
+    end
+
+    status = params[:status]
+    tickets = tickets.select do |ticket|
+      if status
+        ticket.metadata.status == status
```

Comment:
If `metadata` is generic, how do you even know it will always have a `status` variable?

And if you know that every metadata definitely always has a `status` variable, then why is it part of `metadata` and not part of the `Ticket` entity itself?

Reply:
Oh yes, it's not necessary that `status` is there always theoretically. I've added `&`.